### PR TITLE
Employees and Admins controllers, views and models.

### LINKED ISSUE #15: Create Controller and Views of Employees and Admins Class
# REQUIREMENTS
Add the route with which we will start the application, in the config/routes.rb file. In this case, we will start with employees and admins.
 
# Acceptance Criteria
- [x] Controllers Created
- [x] Views Created
- [x] Edited File routes.rb
- [x] Tables created on database

### PR DESCRIPTION
**Pull request by SampabloC**
# Description
- We created controllers, views and models for employees and admins class.
- We add tables on database for every class specified in db folder.
- We linked employee and admin view.

Resolve #15

Co-authored-by: mrivh <m.alicia.riv@gmail.com>

### CODE REVIEW

File: app/controllers/admins_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  end
+
+  # POST /admins or /admins.json
+  def create
+    @admin = Admin.new(admin_params)
+
+    respond_to do |format|
+      if @admin.save
+        format.html { redirect_to @admin, notice: "Admin was successfully created." }
+        format.json { render :show, status: :created, location: @admin }
```
Comment (juliocabrera820) [inline app/controllers/admins_controller.rb:29]: are you building an API REST?

File: app/controllers/admins_controller.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @admin.errors, status: :unprocessable_entity }
```
Comment (juliocabrera820) [inline app/controllers/admins_controller.rb:32]: are you building an API REST?

File: app/controllers/admins_controller.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      end
+    end
+  end
+
+  # PATCH/PUT /admins/1 or /admins/1.json
+  def update
+    respond_to do |format|
+      if @admin.update(admin_params)
+        format.html { redirect_to @admin, notice: "Admin was successfully updated." }
+        format.json { render :show, status: :ok, location: @admin }
```
Comment (juliocabrera820) [inline app/controllers/admins_controller.rb:42]: are you building an API REST?

File: app/controllers/admins_controller.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @admin.errors, status: :unprocessable_entity }
```
Comment (juliocabrera820) [inline app/controllers/admins_controller.rb:45]: are you building an API REST?

File: app/controllers/admins_controller.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      end
+    end
+  end
+
+  # DELETE /admins/1 or /admins/1.json
+  def destroy
+    @admin.destroy
+    respond_to do |format|
+      format.html { redirect_to admins_url, notice: "Admin was successfully destroyed." }
+      format.json { head :no_content }
```
Comment (juliocabrera820) [inline app/controllers/admins_controller.rb:55]: are you building an API REST?

File: app/controllers/admins_controller.rb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_admin
+      @admin = Admin.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def admin_params
+      params.require(:admin).permit(:idAdmin, :Name, :Email, :Password)
```
Comment (juliocabrera820) [inline app/controllers/admins_controller.rb:67]: param names must be in lowercase

**Review by juliocabrera820 — COMMENTED** [review]
Be careful with scaffold, sometimes it generates unneeded files

File: app/assets/stylesheets/admins.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Admins controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (YilKanda) [inline app/assets/stylesheets/admins.scss]: If it doesn't take care, you can delete it.

File: app/assets/stylesheets/employees.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Employees controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (YilKanda) [inline app/assets/stylesheets/employees.scss]: If it doesn't take care, you can delete it.

File: app/helpers/admins_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module AdminsHelper
+end
```
Comment (YilKanda) [inline app/helpers/admins_helper.rb]: If it doesn't take care, you can delete it.

File: app/helpers/employees_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module EmployeesHelper
+end
```
Comment (YilKanda) [inline app/helpers/employees_helper.rb]: If it doesn't take care, you can delete it.

File: app/views/employees/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,3 @@
+<h1>Employees#index</h1>
+<p>Find me in app/views/employees/index.html.erb</p>
+<%= link_to 'Admin', admins_path %>
```
Comment (YilKanda) [inline app/views/employees/index.html.erb]: Add a blank line

File: test/models/employee_test.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class EmployeeTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (YilKanda) [inline test/models/employee_test.rb]: These comments are not necessary.

File: test/models/admin_test.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class AdminTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (YilKanda) [inline test/models/admin_test.rb]: These comments are not necessary.

File: test/controllers/employees_controller_test.rb [hunk 14]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class EmployeesControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (YilKanda) [inline test/controllers/employees_controller_test.rb]: These comments are not necessary.

File: test/controllers/administrators_controller_test.rb [hunk 15]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class AdministratorsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (YilKanda) [inline test/controllers/administrators_controller_test.rb]: These comments are not necessary.

**Review by YilKanda — COMMENTED** [review]
Try to keep the code always clean, following the Ruby style guide, even if you are using some scaffold to generate the codes.

**Review by xFerchoVJ — DISMISSED** [review]
Delete all the things that scaffold generated that you dont gonna use.

**Comment by brayandiazc** [comment]
The colleagues are right about the files that using scaffold generates, but it is understandable that they are starting and it serves as a base, however you can eliminate what does not serve you in the next pull request.
Nice job

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by xFerchoVJ — APPROVED** [review]
LGTM!

**Pull request closed without merging**